docs/release/kubernetes-upgrade.md

```diff
@@ -4,6 +4,37 @@ This document details the K3S kubernetes patch release process.
 # Before You Begin
 You’ll be primarily using git and go. Git can be installed via the local package manager. Make sure Go is installed and configured correctly, utilizing a “gopath”. This can be set via an environment variable called GOPATH. eg. export GOPATH=”${HOME}/go”, typically.
 
+To ensure proper GPG Signing usage, we recommend the following:
+
+ - Disable GPG signing in the git config before running all scripts. The method for disabling GPG signing may vary depending on your specific situation. One approach is to unset the related configuration using the following command:
+
+   ```sh
+   git config --unset-all user.signingkey
+   ```
+
+If you require GPG signing for your tags, configure it based on your specific requirements. Please note that the script assumes that GPG signing is either disabled or configured correctly in the git configuration. 
+
+#####  User Mismatch in Docker Container
+
+When running the script inside a Docker container, there might be a user mismatch between the container user and the git repository directory user. This can result in permission issues when performing Git operations.
+
+To avoid this problem, you can add a safe directory setting in the git configuration by following these steps:
+
+- Create or edit the global git configuration file by running the following command:
+   
+   ```shell
+   git config --global --edit
+  ```
+- In the configuration file, add the following line under the [core] section:
+  ```sh 
+  [core]
+    safelyUseIncompatibleGitCredentialHelper = true
+  ```
+
+This setting ensures that the container user can safely access the Git repository directory without permission issues.
+
+Note: If you are using a specific git configuration file for the project, make sure to add the `safelyUseIncompatibleGitCredentialHelper = true` line in that file instead.
```

Comment:
Unless the user specified otherwise, it should be ok to not do this.

> Each of these “levels” (system, global, local) overwrites values in the previous level, so values in .git/config trump those in [path]/etc/gitconfig, for instance.
https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration

```suggestion
```